### DISCUSSION TITLE
Added: LDM Notes for August 30th, 2021

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#5139] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2021/LDM-2021-08-30.md

## Agenda

1. C# 11 Initial Triage
    1. Generic attributes
    2. List patterns
    3. Static abstracts in interfaces
    4. Declarations under `or` patterns
    5. Records and initialization
    6. Discriminated unions
    7. Params `Span<T>`
    8. Statements as expressions
    9. Expression trees
    10. Type system extensions

I will again caution anyone reading these notes to _not_ try to read the tea leaves about what will be in C# 11. We don't know the answer to that. These are our initial musings, but it will take more work (and meetings with our partners/directors) to determine the relative priorities of issues.

**Comment by huoyaoyuan** [comment]
CRTP has some strange constraint in use, for example https://github.com/dotnet/runtime/discussions/58507 . I'd want you to implement `self` type as soon as possible, to reduce the potential CRTP usages in BCL.

**Comment by DavidArno** [comment]
I don't know about tea leaves. That's an incredible list of features, so I'm holding out for this being an entry from the Nice and Accurate Prophecies of Agnes Nutter.

**Comment by orthoxerox** [comment]
That's a very nice initial list. Glad to see the expression trees might be getting some love. Since it's coming from the EF team, any plans to extend query expressions as well?

**Reply by TonyValenti** [reply]
Oh wow!  That is an amazing list!  If you only get of them done - and I don't care which half - I'll be super thrilled!

**Reply by TonyValenti** [reply]
Is c#11 in the .net 7 timeframe?

**Reply by orthoxerox** [reply]
AFAIK the plan is to release new C# and .Net versions in lockstep every November, so yes.

**Reply by TonyValenti** [reply]
I noticed that required properties are not on this list. Did they get dropped?

**Reply by TheUnlocked** [reply]
Required properties were mentioned in the next LDM notes: https://github.com/dotnet/csharplang/discussions/5141
